### ISSUE TITLE
gitPush deployer fails with github deploy keys

### ISSUE THREAD

**Issue by opennomad** [craftercms/craftercms#3523]
## Describe the bug
Github allows using specific deployment only keys. Those keys are specific to a repo and limited in power, which make them attractive as part of automation. The keys work well from the command line and via the UI. When used as part of the gitPush deployer it results in a repo not found error.

## To Reproduce
1. Create RSA SSH key
2. Add public key to githbu repo deployment keys with write access
3. Add gitPush deployer target using the path to the deploy key
4. See error in /opt/crafter/logs/deployer/crafter-deployer.out

## Expected behavior
There should not be an error and the push should succeed

## Logs
```
org.craftercms.deployer.api.exceptions.DeployerException: Execution of git push failed; nested exception is org.eclipse.jgit.api.errors.TransportException:  ssh://git@github-aliase/ORG/REPO.git: ERROR: Repository not found.

        at org.craftercms.deployer.impl.processors.GitPushProcessor.doPush(GitPushProcessor.java:97)
        at org.craftercms.deployer.impl.processors.GitPushProcessor.doExecute(GitPushProcessor.java:69)
        at org.craftercms.deployer.impl.processors.AbstractMainDeploymentProcessor.execute(AbstractMainDeploymentProcessor.java:70)
        at org.craftercms.deployer.impl.DeploymentPipelineImpl.executeProcessors(DeploymentPipelineImpl.java:76)
        at org.craftercms.deployer.impl.DeploymentPipelineImpl.execute(DeploymentPipelineImpl.java:68)
        at org.craftercms.deployer.impl.TargetImpl$DeploymentTask.run(TargetImpl.java:242)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: org.eclipse.jgit.api.errors.TransportException:  ssh://git@github-aliase/ORG/REPO.git: ERROR: Repository not found.

        at org.eclipse.jgit.api.PushCommand.call(PushCommand.java:180)
        at org.craftercms.deployer.utils.GitUtils.push(GitUtils.java:181)
        at org.craftercms.deployer.impl.processors.GitPushProcessor.doPush(GitPushProcessor.java:81)
        ... 10 common frames omitted
Caused by: org.eclipse.jgit.errors.NoRemoteRepositoryException: ssh://git@github-aliase/ORG/REPO.git: ERROR: Repository not found.

        at org.eclipse.jgit.transport.TransportGitSsh.cleanNotFound(TransportGitSsh.java:218)
        at org.eclipse.jgit.transport.TransportGitSsh$SshPushConnection.<init>(TransportGitSsh.java:372)
        at org.eclipse.jgit.transport.TransportGitSsh.openPush(TransportGitSsh.java:176)
        at org.eclipse.jgit.transport.PushProcess.execute(PushProcess.java:160)
        at org.eclipse.jgit.transport.Transport.push(Transport.java:1344)
        at org.eclipse.jgit.api.PushCommand.call(PushCommand.java:169)
        ... 12 common frames omitted

```

### Version

Crafter Studio

Studio Version Number: 3.0.20E-c38ab4

Build Number: c38ab472a84d1b396825f45c2ef96ff1cec985e3

Build Date/Time: 01-17-2019 16:57:42 -0800

### OS
Crafter official AMI

**Comment by jdrossl** [comment]
hi @opennomad , I was able to use the push processor with a deploy key, my configuration looks like this:
```
- processorName: gitPushProcessor
  remoteRepo:
    url: git@github.com:jdrossl/push-test.git
    branch: master
    ssh:
      privateKey:
        path: /path/to/file
        passphrase: ***
```

I noticed in the stacktrace you provided the URL might have an issue:

```ssh://git@github-aliase/ORG/REPO.git```

should be:

```ssh://git@github-aliase:ORG/REPO.git```

Please try changing the URL and let us know if that fixes the issue.

**Comment by opennomad** [comment]
@jdrossl , that was attempt number 3 at making it work. I had the plain github provided target (i.e. `git@github:org/repo`) first. Then I tried with the protocol as you suggested (i.e. `ssh://`git@github:org/repo`) and that failed. Then I found https://www.codeaffine.com/2014/12/09/jgit-authentication/, where the URL in the example swaps the `:` with `/` and that's what I ended up using for the bug report.

I ended up wrting a simple cron script that does the push. I'll try and find some time to retest as you suggest.
